### ISSUE TITLE
Add an explorer/title menu contribution point

### ISSUE THREAD

**Issue by eamodio** [microsoft/vscode#108271]
Currently extensions can't contribute to the Explorer's title menu

![image](https://user-images.githubusercontent.com/641685/95380025-256d7d00-08b4-11eb-9ea0-6b761fc3110b.png)

It would be great to allow this. For example it would allow for a better UX in the [Toggle Excluded Files](https://marketplace.visualstudio.com/items?itemName=eamodio.toggle-excluded-files) to add the toggle there rather than omni-present in the statusbar (See #103536 for an ask)

**Comment by isidorn** [comment]
@eamodio I think this is a fair request.
However, I would be worried that too many actions end up being there which would lead to too many details.

So for now no immediate plans but let's leave in the backlog and I am open to allow this in the future if there are more good use cases.

**Comment by eamodio** [comment]
😢

**Comment by gjsjohnmurray** [comment]
> However, I would be worried that too many actions end up being there which would lead to too many details.

@isidorn that could be mitigated by implementing a '...' secondary actions menubutton which would appear if at least one contribution to `explorer/title` specified a group that disqualifies it from being a button on the title. The `scm/title' menu operates this way.

**Comment by JacksonKearl** [comment]
An "Add config files to project" extension would be a good client of this menu, probably others. Following SCM view's lead as @gjsjohnmurray suggests sounds reasonable.

**Comment by jaredbroad** [comment]
🙏 Please reconsider for the backlog @vscode-triage-bot

**Comment by JacksonKearl** [comment]
`On Deck` is higher than `Backlog`

**Comment by jaredbroad** [comment]
Awesome thank you @JacksonKearl! Maybe for the `December 2021` milestone then 😁 🙏 

>'...' secondary actions menu button

It would be nice if this operated as an overflow -- e.g. 3 buttons can appear in the bar but on the 4th it appears in the `...` section.

**Comment by JacksonKearl** [comment]
Explorer panel only. Some of the other panels already support contributions, such as SCM.

**Comment by jaredbroad** [comment]
Do custom panels support buttons @JacksonKearl? We couldn't find anything about it.

**Comment by JacksonKearl** [comment]
@jaredbroad yes, I know GitLens's panels have actions, I assume the code is pretty much the same as core's panels.

**Comment by dmalan** [comment]
Just to add to the mix with a use case we have in mind in educational contexts, Codespaces tends to assume that a user's default workspace is `/workspaces/foo`, but users can open a specific subfolder therein, as file **File > Open Folder**. E.g., if a user opens `bar`, they'll find themselves at a URL like `https://{name}.github.dev/?folder=%2Fworkspaces%2Ffoo%2Fbar`. Once there, though, it's not particularly obvious in GUI how to pop the stack to escape `bar`, except via, e.g., **File > Open Folder** again or by changing the URL to `https://{name}.github.dev/?folder=%2Fworkspaces%2Ffoo`. 

Via an extension, then, we're hoping to add a "parent folder" icon to Explorer that takes students to `..` with a single click or perhaps even a "home folder" icon that takes students back to `/workspaces/$RepositoryName`.

We can contribute such now via `view/title`, but the icons then end up appearing in multiple views, not just Explorer. Ideally, it'd be possible to scope such to that specific view, a la:

```json
{
  "command": "extension.home",
  "group": "navigation",
  "when": "view == explorer"
}
```

Many thanks!

CC @rongxin-liu @jkeech

**Comment by lramos15** [comment]
> We can contribute such now via `view/title`, but the icons then end up appearing in multiple views, not just Explorer. Ideally, it'd be possible to scope such to that specific view, a la:

The explorer already has 4 icons which is where we normally cap number of displayed icons in a view panel. As @gjsjohnmurray stated once this is implemented these contributions are likely to exist as a `...` menu within the explorer. I'm curious if you feel this will hinder student's discovery of the action
